dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ParamTypesResolver.java

```diff
@@ -158,23 +162,53 @@ private ParameterEncoderType multimapDefaultEncoder(EndpointName endpointName, V
                 .build();
     }
 
+    private Optional<ParameterEncoderType> getPathParameterEncoder(
+            EndpointName endpointName,
+            VariableElement variableElement,
+            AnnotationReflector annotationReflector,
+            EncoderTypeAndMethod encoderTypeAndMethod,
+            EncoderTypeAndMethod listEncoderTypeAndMethod) {
+        Optional<TypeName> encoderTypeName =
+                annotationReflector.getFieldMaybe("encoder", TypeMirror.class).map(TypeName::get);
+
+        Optional<TypeName> listEncoderTypeName = annotationReflector
+                .getFieldMaybe("listEncoder", TypeMirror.class)
+                .map(TypeName::get);
+
+        if (encoderTypeName.isPresent() && listEncoderTypeName.isPresent()) {
+            throw new SafeRuntimeException("Only one of encoder and listEncoder can be set for " + endpointName.get());
```

Comment:
Perhaps `context.reportError` instead of throwing, we can return empty optional since we're failing the build.